js-packages/@prestojs/routing/src/UrlPattern.ts

```diff
@@ -88,9 +123,14 @@ export default class UrlPattern {
      * @param kwargs The arguments to resolve in the pattern. For example if the pattern was
      * `/users/:id/:section?` you would need to pass at least `id` and optionally `section`:
      * `pattern.resolve({id: 5})`
+     * @param options Options to use for this resolution. Any options provided here will override an option
+     * of the same name provided to the constructor.
      */
-    resolve(kwargs: {} = {}, options: ResolveOptions = {}): string {
-        const { query } = options;
+    resolve(kwargs: {} = {}, options: UrlPatternResolveOptions = {}): string {
+        let { baseUrl, mergeQuery = true } = { ...this.resolveOptions, ...options };
+        const query = mergeQuery
+            ? { ...this.resolveOptions.query, ...options.query }
+            : options.query || this.resolveOptions.query;
```

Comment:
i think its worth a mention (in the example above) on the way to suppress default constructor query, ie, i'd have expected {query: false, mergeQuery: false} is a valid way to disable default query but that wont work